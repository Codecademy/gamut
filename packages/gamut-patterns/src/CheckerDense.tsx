import { Box } from '@codecademy/gamut';
import styled from '@emotion/styled';
import React from 'react';

import { PatternProps } from './types';

const Svg = styled.svg`
  width: 100%;
  height: 100%;
`;

export const CheckerDense: React.FC<PatternProps> = ({ ...props }) => (
  <Box {...props}>
    <Svg aria-hidden>
      <pattern
        id="checkerDense"
        x="0"
        y="0"
        width="4"
        height="4"
        patternUnits="userSpaceOnUse"
      >
        <rect width="1" height="1" fill="currentColor" />
        <rect x="2" y="2" width="1" height="1" fill="currentColor" />
      </pattern>
      <rect
        x="0"
        y="0"
        width="100%"
        height="100%"
        fill={`url(#checkerDense)`}
      />
    </Svg>
  </Box>
);
