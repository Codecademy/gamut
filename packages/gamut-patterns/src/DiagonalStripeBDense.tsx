import { Box } from '@codecademy/gamut';
import styled from '@emotion/styled';
import React from 'react';

import { PatternProps } from './types';

const Svg = styled.svg`
  width: 100%;
  height: 100%;
`;

export const DiagonalStripeBDense: React.FC<PatternProps> = ({ ...props }) => (
  <Box {...props}>
    <Svg aria-hidden>
      <pattern
        id="diagonalStripeBDense"
        x="0"
        y="0"
        width="4"
        height="4"
        patternUnits="userSpaceOnUse"
      >
        <g clipPath="url(#clipDiagonalStripeBDense)">
          <rect width="4" height="4" fill="white" />
          <rect y="3" width="1" height="1" fill="currentColor" />
          <rect x="1" y="2" width="1" height="1" fill="currentColor" />
          <rect x="2" y="1" width="1" height="1" fill="currentColor" />
          <rect x="3" width="1" height="1" fill="currentColor" />
        </g>
        <defs>
          <clipPath id="clipDiagonalStripeBDense">
            <rect width="4" height="4" fill="white" />
          </clipPath>
        </defs>
      </pattern>
      <rect
        x="0"
        y="0"
        width="100%"
        height="100%"
        fill={`url(#diagonalStripeBDense)`}
      />
    </Svg>
  </Box>
);
