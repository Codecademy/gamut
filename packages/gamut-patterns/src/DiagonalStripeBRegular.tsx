import { Box } from '@codecademy/gamut';
import styled from '@emotion/styled';
import React from 'react';

import { PatternProps } from './types';

const Svg = styled.svg`
  width: 100%;
  height: 100%;
`;

export const DiagonalStripeBRegular: React.FC<PatternProps> = ({
  ...props
}) => (
  <Box {...props}>
    <Svg aria-hidden>
      <pattern
        id="diagonalStripeBRegular"
        x="0"
        y="0"
        width="8"
        height="8"
        patternUnits="userSpaceOnUse"
      >
        <g clipPath="url(#clipDiagonalStripeBRegular)">
          <rect width="8" height="8" fill="white" />
          <rect y="7" width="1" height="1" fill="currentColor" />
          <rect x="1" y="6" width="1" height="1" fill="currentColor" />
          <rect x="2" y="5" width="1" height="1" fill="currentColor" />
          <rect x="3" y="4" width="1" height="1" fill="currentColor" />
          <rect x="4" y="3" width="1" height="1" fill="currentColor" />
          <rect x="5" y="2" width="1" height="1" fill="currentColor" />
          <rect x="6" y="1" width="1" height="1" fill="currentColor" />
          <rect x="7" width="1" height="1" fill="currentColor" />
        </g>
        <defs>
          <clipPath id="clipDiagonalStripeBRegular">
            <rect width="8" height="8" fill="white" />
          </clipPath>
        </defs>
      </pattern>
      <rect
        x="0"
        y="0"
        width="100%"
        height="100%"
        fill={`url(#diagonalStripeBRegular)`}
      />
    </Svg>
  </Box>
);
