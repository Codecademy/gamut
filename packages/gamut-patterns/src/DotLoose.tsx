import { Box } from '@codecademy/gamut';
import styled from '@emotion/styled';
import React from 'react';

import { PatternProps } from './types';

const Svg = styled.svg`
  width: 100%;
  height: 100%;
`;

export const DotLoose: React.FC<PatternProps> = ({ ...props }) => (
  <Box {...props}>
    <Svg aria-hidden>
      <pattern
        id="dotLoose"
        x="0"
        y="0"
        width="16"
        height="16"
        patternUnits="userSpaceOnUse"
      >
        <rect width="0.5" height="0.5" fill="currentColor" fillOpacity="0.5" />
        <rect
          y="1"
          width="0.5"
          height="0.5"
          fill="currentColor"
          fillOpacity="0.5"
        />
        <rect y="0.5" width="0.5" height="0.5" fill="currentColor" />
        <rect
          x="1"
          width="0.5"
          height="0.5"
          fill="currentColor"
          fillOpacity="0.5"
        />
        <rect
          x="1"
          y="1"
          width="0.5"
          height="0.5"
          fill="currentColor"
          fillOpacity="0.5"
        />
        <rect x="1" y="0.5" width="0.5" height="0.5" fill="currentColor" />
        <rect x="0.5" width="0.5" height="0.5" fill="currentColor" />
        <rect x="0.5" y="1" width="0.5" height="0.5" fill="currentColor" />
        <rect x="0.5" y="0.5" width="0.5" height="0.5" fill="currentColor" />
      </pattern>
      <rect x="0" y="0" width="100%" height="100%" fill={`url(#dotLoose)`} />
    </Svg>
  </Box>
);
