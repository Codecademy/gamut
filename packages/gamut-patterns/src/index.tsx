export * from './patterns/checker';
export * from './patterns/diagonalA';
export * from './patterns/diagonalB';
export * from './patterns/dot';
export * from './patterns/ex';
export * from './patterns/flower';
export * from './patterns/grid';
export * from './patterns/herringbone';
export * from './patterns/rain';
export * from './patterns/stripe';
export * from './types';
