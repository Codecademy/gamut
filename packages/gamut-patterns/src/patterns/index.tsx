export * from './CheckerDense';
export * from './CheckerLoose';
export * from './CheckerRegular';
export * from './DiagonalADense';
export * from './DiagonalALoose';
export * from './DiagonalARegular';
export * from './DiagonalBDense';
export * from './DiagonalBLoose';
export * from './DiagonalBRegular';
export * from './DotDense';
export * from './DotLoose';
export * from './DotRegular';
export * from './ExDense';
export * from './ExLoose';
export * from './ExRegular';
export * from './FlowerDense';
export * from './FlowerLoose';
export * from './FlowerRegular';
export * from './GridDense';
export * from './GridLoose';
export * from './GridRegular';
export * from './Herringbone';
export * from './RainDense';
export * from './RainLoose';
export * from './RainRegular';
export * from './StripeDense';
export * from './StripeLoose';
export * from './StripeRegular';
