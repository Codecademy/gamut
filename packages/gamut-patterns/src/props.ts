import { styledOptions, system } from '@codecademy/gamut-styles';
import { StyleProps, variance } from '@codecademy/variance';
import styled from '@emotion/styled';

const patternStyles = variance.compose(
  system.layout,
  system.positioning,
  system.space
);

export type PatternStyleProps = StyleProps<typeof patternStyles>;
export interface PatternProps
  extends Omit<React.SVGProps<SVGSVGElement>, keyof PatternStyleProps>,
    PatternStyleProps {
  /**
   * A suffix added to the end of the unique generated ID for the icon. This is useful if you have multiple of the same icon on the page and need to pass accessibility guidelines.
   */
  titleId?: string;
  title?: string;
  ref?: React.Ref<SVGSVGElement>;
}

export const Svg = styled(
  'svg',
  styledOptions<'svg'>()
)<PatternProps>(patternStyles);

Svg.defaultProps = {
  width: '100%',
  height: '100%',
};
