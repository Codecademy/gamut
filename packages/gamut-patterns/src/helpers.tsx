import { CheckerDense } from './CheckerDense';
import { CheckerLoose } from './CheckerLoose';
import { CheckerRegular } from './CheckerRegular';
import { DiagonalStripeBDense } from './DiagonalStripeBDense';
import { DiagonalStripeBLoose } from './DiagonalStripeBLoose';
import { DiagonalStripeBRegular } from './DiagonalStripeBRegular';
import { DotLoose } from './DotLoose';

export const getPattern = (name: string) => {
  switch(name) {
    case 'CheckerDense':
      return CheckerDense;
    case 'CheckerLoose':
      return CheckerLoose;
    case 'CheckerRegular':
      return CheckerRegular;
    case 'DiagonalStripeBDense':
      return DiagonalStripeBDense;
    case 'DiagonalStripeBLoose':
      return DiagonalStripeBLoose;
    case 'DiagonalStripeBRegular':
      return DiagonalStripeBRegular;
    case 'DotLoose':
      return DotLoose;
    default:
      return DiagonalStripeBLoose;
  }
}
