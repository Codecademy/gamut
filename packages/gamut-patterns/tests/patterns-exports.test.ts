import * as patterns from '../src/index';

test('Patterns Exported Keys 1', () => {
  const exportedNames = Object.keys(patterns).sort((a, b) =>
    a.toLowerCase().localeCompare(b.toLowerCase())
  );
  expect(exportedNames).toMatchSnapshot();
});
