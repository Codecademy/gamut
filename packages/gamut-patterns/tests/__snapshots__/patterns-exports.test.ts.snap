// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patterns Exported Keys`] = `
[
  "CheckerDense",
  "CheckerLoose",
  "CheckerRegular",
  "DiagonalADense",
  "DiagonalALoose",
  "DiagonalARegular",
  "DiagonalBDense",
  "DiagonalBLoose",
  "DiagonalBRegular",
  "DotDense",
  "DotLoose",
  "DotRegular",
  "ExDense",
  "ExLoose",
  "ExRegular",
  "FlowerDense",
  "FlowerLoose",
  "FlowerRegular",
  "GridDense",
  "GridLoose",
  "GridRegular",
  "Herringbone",
  "RainDense",
  "RainLoose",
  "RainRegular",
  "StripeDense",
  "StripeLoose",
  "StripeRegular",
]
`;
